UI/FrontEnd.py

```diff
@@ -0,0 +1,85 @@
+from tkinter import Frame, Label, Button, OptionMenu, Tk, StringVar, RAISED
+from tkinter.colorchooser import askcolor
+from VolumeLogic import ChangedValue, VolumeMixer
+from SystemConnectors import WindowsConnection
+from functools import partial
+
+
+class FrontEnd:
+    """Handles the Settings UI"""
+    
+    def __init__(self, volume_mixer):
+        """
+        Initializes FrontEnd
+
+        :param volume_mixer: Volume Mixer used to communicate with rest of application
+        """
+        self.volume_mixer = volume_mixer
+        self.current_configuration = self.volume_mixer.applications
+        self.open_apps = self.volume_mixer.get_running_applications()
+
+    def open_color_dialog(self, knob):
+        """
+        Opens color picker and passes selected color to Volume Mixer
+
+        :param knob: Knob number
+        """
+        color = askcolor()[0]
+        color = [int(x) for x in color]
+        self.volume_mixer.modify_application(knob, ChangedValue.COLOR, color)
+
+   
+    def get_apps(self):
+        """
+        Gets apps for dropdown, occurs every time dropdown is opened
+
+        :return: list of app names with last 4 characters removed
+        """
+        self.open_apps = self.volume_mixer.get_running_applications()
+        return [app.name for app in self.open_apps]
+
+    
+    def update_app(self, knob, app_name):
+        """
+        Sends app seleced from dropdown to volumemixer, tied with knob
+
+        :param knob Knob number 
+        :param app_name: name of selected app
+        """
+        for i in self.open_apps:
+            if(i.name.find(app_name)):
+                app = i
+                break
+        self.volume_mixer.set_application(knob, app)
+  
+    
+    def launch(self):
+        """
+        Launches UI
+        """
+        root = Tk()
+        root.title("Bricxer")
+        main_frame = Frame(root, bg='Purple')
+        main_frame.grid(row=0, column=0, padx=10, pady=5)
+        
+        knobs = self.volume_mixer.applications
+        for knob in range(len(knobs)):
```

Comment:
Change name of knobs to applications since you are getting application objects, not knobs. The name of knob should also change, but that is a bit more complicated.

* If the current for loop is going to remain the same, change the name to index.
* If at some point we are going to decide to load configuration, the application information will need to be known when creating the different UI objects. `for index, app in enumerate(applications)` should be used instead.

If knobs is only set to get the applications, I would also consider making it a 1 liner: `for knob in range(len(self.volume_mixer.applications))`. And then the renaming mentioned above.